File: informatika_ismeretek/emelt/2019_may/java/Nasa.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
import java.util.stream.*;

public class Nasa {

    public static void main(String[] args) throws IOException {
        var keresek = new ArrayList<Keres>();
        for(var line : Files.readAllLines(Path.of("NASAlog.txt"))) {
            keresek.add(new Keres(line));
        }
        
        System.out.println("5. Feladat: Kérések száma: " + keresek.size());
        
        var osszmeret = 0;
        for(var keres : keresek) {
            osszmeret += keres.byteMeret();
        }
        
        System.out.println("6. Feladat: Összméret: " + osszmeret + " byte");
        
        var domainesek = 0;
        for(var keres : keresek) {
            if(keres.domain()){
                ++domainesek;
            }
        }
        
        System.out.printf("8. Feladat: Domaines kérések: %.2f%%\n", ((float) domainesek) / keresek.size() * 100);
        System.out.println("9. Feladat:");
        
        var stat = new HashMap<String, Integer>();
        for(var keres : keresek) {
            stat.put(keres.httpKod, stat.getOrDefault(keres.httpKod, 0) + 1);
        }
        
        for(var entry : stat.entrySet()) {
            System.out.println("    " + entry.getKey() + ": " + entry.getValue() + " db");
        }
    }
}

File: informatika_ismeretek/emelt/2019_may/python/Nasa.py
from collections import Counter

class Keres:
    def __init__(self, line: str):
        split = line.split("*")
        utolsoSzokozIndex = split[3].find(" ")

        self.cim = split[0]
        self.datumIdo = split[1]
        self.keres = split[2]
        self.httpKod = split[3][:utolsoSzokozIndex]
        self.meret = split[3][utolsoSzokozIndex + 1:]

    def byteMeret(self):
        return 0 if self.meret == "-\n" else int(self.meret)

    def domain(self):
        return str.isalpha(self.cim[len(self.cim) - 1])

with open("NASAlog.txt", "r") as file:
    keresek = [ Keres(k) for k in file.readlines() ]

osszmeret = sum(k.byteMeret() for k in keresek)
domainesek = sum(1 for k in keresek if k.domain())

print(f"5. Feladat: Kérések száma: {len(keresek)}")
print(f"6. Feladat: Összméret: {osszmeret} byte")
print("8. Feladat: Domaines kérések: %.2f%%" % (domainesek / len(keresek) * 100))

stat = Counter(k.httpKod for k in keresek)
for key, value in stat.items():
    print(f"    {key}: {value} db")